src/Components/Browser.JS/src/Platform/Circuits/DefaultReconnectDisplay.ts

```diff
@@ -0,0 +1,34 @@
+import { ReconnectDisplay } from "./ReconnectDisplay";
+import { AutoReconnectCircuitHandler } from "./AutoReconnectCircuitHandler";
+export class DefaultReconnectDisplay implements ReconnectDisplay {
+  modal: HTMLDivElement;
+  message: HTMLHeadingElement;
+  button: HTMLButtonElement;
+  addedToDom: boolean = false;
+  constructor(private document: Document) {
+    this.modal = this.document.createElement('div');
+    this.modal.id = AutoReconnectCircuitHandler.DialogId;
+    this.modal.style.cssText = 'position:fixed;top:0;right:0;bottom:0;left:0;z-index:1000;display:none;overflow:hidden;background-color:#fff;opacity:0.8;text-align:center;font-weight:bold';
+    this.modal.innerHTML = '<h5 style="margin-top: 20px"></h5><button style="margin:5px auto 5px">Retry?</button>';
+    this.message = this.modal.querySelector('h5')!;
+    this.button = this.modal.querySelector('button')!;
+
+    this.button.addEventListener('click', () => window['Blazor'].reconnect());
```

Comment:
Shouldn't this be done automatically after a timeout instead of making the user click a button? Is this in addition to that?

Reply:
Yeah. Once we fail auto-reconnects you get to manually retry using this button.